### PR TITLE
AddSmoothScrollingFunctionnality

### PR DESCRIPTION
**Pull request by mickaelalbertus**
Add the smooth scrolling functionnality. There is a bug in Gtk: we have to choose between Smooth scroll mask or not and if we choose it there is only the direction GDK_SCROLL_SMOOTH that is caught. So this part of code is different from evince and didn't manage the other direction because they never happened. Maybe this bug is corrected with the newer version of Gtk?

### CODE REVIEW

**Comment by clefebvre** [comment]
Hi @mickaelalbertus,

I don't understand the issue. Can you explain in layman terms how to observe the problem?

Also, which versions of GTK are at play here?

**Comment by mickaelalbertus** [comment]
Sorry I wasn’t understandable. This add the smooth scrolling functionality to xreader (when you scroll the document).

I said at the same time, the code is different from Evince project because there is a knowing GTK bug when I add the smooth scrolling mask. Directions GDK_SCROLL_UP, GDK_SCROLL_LEFT, etc... are not recognized so it's for that reason that I erase them from the code and I manage it with event->delta_x and event->delta_y.

**Comment by clefebvre** [comment]
when you say smooth-scrolling, you mean the fact of scrolling without jumps by right-clicking an empty area of the scrollbar and moving left/right to scroll faster?

**Comment by clefebvre** [comment]
How do you reproduce this exactly? With a mouse wheel or touch screen? Can you describe the steps/actions you're taking?

I've been trying to scroll for a while now... using both versions of the code.. I can't see any difference :))

**Comment by mickaelalbertus** [comment]
I thought it was clear sorry :)
By smooth scrolling I meant that when you scroll the document it's smooth. For example if you try to scroll a lot with this version and you stop to scroll, the document continues to scroll smoothly before stopping and it doesn't stop to scroll immediately. Do you see what I mean? 

This change is more convenient for users which uses tablets I think.

**Comment by clefebvre** [comment]
Hi @mickaelalbertus, careful.. your latest changes on this branch (30 min ago or so) break some of the changes from your other PRs and the indentation as well.

**Comment by mickaelalbertus** [comment]
Yes I hadn't paid attention! I'm still learning the mechanism of git, sorry! :)

**Comment by clefebvre** [comment]
Hi @mickaelalbertus, please fix the trailing spaces.

**Comment by mickaelalbertus** [comment]
Done!

**Comment by clefebvre** [comment]
Hi @mickaelalbertus, this is still has many trailing spaces :)

You can see them here in the diff tab of the PR, they show up in green.

**Comment by mickaelalbertus** [comment]
Sorry for the trailing spaces. I won't never do it again since I've installed an automatic plugin :)
